### ISSUE TITLE
`FIX` rules with expire date or deadlines

### ISSUE THREAD

**Issue by edpyt** [astral-sh/ruff#16963]
### Summary

I want to have a feature to write `# TODO:` comments with deadlines.  
Something like:  
```python
s: str = "1"  # TODO(<=30.03.2025): rewrite to int
```

And if today date is 01.04.2025, then in next `ruff check` i want to see an error like:  
```bash
Today is 01.04.2025 you need to fix this!
```

**Comment by edpyt** [comment]
thanks for the reply!

> It's also not clear what a user would do 

if it will be released as a separate rule (e.g. `FIX005`), then probably:
* ignore this rule.  
  (do you think this will be inconvenient?)  
* update dates/fix todos.  

> because fixing a TODO often takes a fair bit of time

I think this is a matter of habit. user should set a correct deadline for completing code changes

also, I found that ESLint has a plugin with similar functionality: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/expiring-todo-comments.md

##### offtop:
> My most likely action would be to bump the TODO date.

normally me, always bump the deadlines dates :D

**Comment by MichaReiser** [comment]
That's an excellent point. It breaks many assumptions of how ruff works
